lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -886,6 +886,11 @@ SPIRVValue *LLVMToSPIRV::transValueWithoutDecoration(Value *V,
           BM->addLoopMergeInst(Br->getFalseLabel()->getId(), // Merge Block
                                BB->getId(),                  // Continue Target
                                LoopControl, Parameters, SuccessorTrue);
+        } else {
+          // For unstructured loop we add a special loop control instruction.
+          // Simple example of unstructured loop is an infinite loop, that has
+          // no terminate instruction.
+          BM->addLoopControlINTELInst(LoopControl, Parameters, SuccessorTrue);
```

Comment:
This depends on #244 as this is an extension

not sure how should this handled, i.e. wait for #244 to land or follow up once it landed

Reply:
I think I will commit it first and then rebase on #244, preparing a patch that Alexey can include in his PR.